frontend/lib/hooks/use-local-storage.ts

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useState } from 'react';
+
+export function useLocalStorage<T>(key: string, defaultValue: T) {
+  const [value, setValue] = useState<T>(() => {
+    if (typeof window === 'undefined') {
+      return defaultValue;
+    }
+
+    const item = window.localStorage.getItem(key);
```

Comment:
Consider sessionStorage instead of localStorage because API keys are sensitive credentials that shouldn't persist indefinitely. 

The sessionStorage automatically clears when the browser tab closes, reducing the risk if a user forgets to sign out or is using a shared/public computer (Common in academic environments).

Reply:
In that case I think it's probably better to not persist the API token in the browser at all, just always ask the user to provide it every time. wyt?

Reply:
I think it would be fine if we just keep it in the sessionStorage, since we do not have any user-generated content.